src/agents/coordinator_agent.py

```diff
@@ -0,0 +1,172 @@
+        template.to = f"{self.jid}@{DEFAULT_HOST}"
+        template.set_metadata("performative", "request")
+        template.set_metadata("action", "modify-reservation")
+        return template
+
+    def _prepare_parking_availability_template(self):
+        template = Template()
+        template.to = f"{self.jid}@{DEFAULT_HOST}"
+        template.set_metadata("performative", "inform")
+        template.set_metadata("action", "parking-availability")
+        return template
+
+    def _prepare_reservation_response_template(self):
+        template = Template()
+        template.to = f"{self.jid}@{DEFAULT_HOST}"
+        template.set_metadata("performative", "inform")
+        template.set_metadata("action", "reservation-response")
+        return template
+
+    def _calculate_distance(self, x_user, y_user, x_parking, y_parking):
+        return ((x_user - x_parking) ** 2 + (y_user - y_parking) ** 2) ** 0.5
+
+    def _consolidate_offers_per_user(self, user) -> ConsolidatedOffers:
+        dest_x, dest_y = self._per_user_data[user]["destination"]
+        free_parkings = [parking for parking in self._per_user_data[user]["parkings"] if parking.available]
+        offers = []
+        for parking in free_parkings:
+            offer = Offer(parking_id=parking.parking_id, price=parking.parking_price,
+                          distance=self._calculate_distance(dest_x, dest_y, parking.parking_x, parking.parking_y))
+            offers.append(offer)
+        offers = sorted(offers, key=lambda x: x.distance)
+        offers = offers[:3]
+        return ConsolidatedOffers(offers=offers)
+
+    class CheckParkingOffers(CyclicBehaviour):
+        """Behaviour for checking parking offers"""
+
+        def _check_if_request_is_within_region(self, request_x, request_y):
+            return self.agent._x_min <= request_x < self.agent._x_max and self.agent._y_min <= request_y < self.agent._y_max
+
+        async def run(self):
+            msg = await self.receive(timeout=1)
+            if msg:
+                check_offers_message = CheckOffers(**json.loads(msg.body))
+                self.agent._per_user_data[check_offers_message.sender] = {
+                    "parkings": [], "destination": (check_offers_message.x, check_offers_message.y)}
+                if self._check_if_request_is_within_region(check_offers_message.x, check_offers_message.y):
+                    check_parking = str(CheckParking(check_offers_message.time_start,
+                                        check_offers_message.time_stop).dict())
+                    for jid in self.agent._parking_agents_jids:
+                        # user_jid@host in message.thread, need in response from parking agent
+                        to_send = Message(to=jid, body=check_parking, thread=check_offers_message.sender, metadata={
+                                          "performative": "query-ref", "action": "check-parking"})
+                        await self.send(to_send)
```

Comment:
How about adding some message that the request is outside of the `RefionalCoordinator`'s area?

Reply:
In my humble opinion, UserAgent should not receive any messages if request is not in currect RegionalCoordinator area, because in fact, UserAgent do not know which RegionalCoordinator responds for desired localization.

Reply:
Sounds good.
How about adding some message if there isn't any RegionalCoordinator in the Requested Area? e.g. a timeout for the response from any RegionalCoordinator and if none comes, then we can display a message to the user

Reply:
I think, that it is fundamental that every area has RegionalCoordinator.

Reply:
Moreover, it is not possible to check if someone reply for the CheckOffers. I do not see any solution for that.
Thats why, we should make assumption that whole area is covered with Coordinators

Reply:
it was decided that:
- that it is fundamental that every area has RegionalCoordinator
- there has to be a timeout on the user site to prevent endless waiting, e.g. in case of the agents failure